bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateService.java

```diff
@@ -327,6 +325,11 @@ private ProgressCallbackImpl getProgressCallback(ThingUID thingUID) {
         return progressCallbackMap.get(thingUID);
     }
 
+    private Firmware getLatestSuitableFirmware(Thing thing) {
+        return firmwareRegistry.getFirmwares(thing.getThingTypeUID()).stream()
```

Comment:
Question: does getFirmwares return a sorted list, so that findFirst is taking the correct one (latest one)?

Reply:
Yes, it does (not a list, but a collection), cf. the Javadoc of the method: _The collection is sorted in descending order, i.e. the latest firmware will be the first element in the collection._